internal/collect/collect.go

```diff
@@ -23,49 +23,53 @@ func CollectData(config *config.Config) {
 			config.Logrus.WithError(err).Error("failed to build tag cache")
 		}
 	}
-	config.Logrus.WithField("seconds", config.Uptime().Seconds()).Debug("after collecting tags")
+	config.Logrus.WithField("seconds", config.Uptime()).Debug("after collecting tags")
 
 	err := Datacenters(config)
 	if err != nil {
 		return
 	}
-	config.Logrus.WithField("seconds", config.Uptime().Seconds()).Debug("after collecting dc data")
+	config.Logrus.WithField("seconds", config.Uptime()).Debug("after collecting dc data")
 
+	if len(config.Datacenters) == 0 {
+		config.Logrus.Debug("returning early. no datacenter was collected.")
```

Comment:
i think that if this happened is probably due to a bad filtering specification. What do you think to elevate this to error so it appears clearly in the agent log?

Reply:
yeah, makes sense

Reply:
do you think we should stop the integration completely if we do not get any datacenter?

Reply:
yes i think non of the collected data will make sense without the datacenter. Anyway the agent will wait until the next interval to lunch it again , but it will be clear in the agent logs (info mode) that the collection of the integration have returned no content.